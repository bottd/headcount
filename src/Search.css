form {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

form * {
  font-size: 24px;
}

input {
  width: 40%;
}
