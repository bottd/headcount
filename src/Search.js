import React, { Component } from 'react';
import './Search.css';

class Search extends Component {
  constructor() {
    super();
    this.state = {
      search: ''
    }
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.filterCards(this.state.search);
  }

  updateValue = (e) => {
    this.setState({search: e.target.value});
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          value={this.state.search}
          placeholder='Search by name'
          onChange={this.updateValue}
        />
        <button>Search</button>
      </form>
    )
  }
}

export default Search;
