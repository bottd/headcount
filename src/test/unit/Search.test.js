import React from 'react';
import { shallow, mount } from 'enzyme';
import Search from '../../Search';

describe('Search', () => {
  it('Renders with one text input and 1 button', () => {
    const wrapper = shallow(<Search/>);
    expect(wrapper).toMatchSnapshot()
  });

  it('Should have a state for the search value', () => {
    const wrapper = shallow(<Search/>);
    expect(wrapper.state().search).toBe('');
  });

  it('Should update state onChange', () => {
    const wrapper = shallow(<Search/>);
    const mockEvent = { target: { value: 'Colorado'}};
    wrapper.instance().updateValue(mockEvent);
    expect(wrapper.state().search).toBe('Colorado');
  });

  it('Should call filterCards onSubmit', () => {
    const filterCards = jest.fn();
    const wrapper = shallow(<Search filterCards={filterCards}/>);
    const mockEvent = {preventDefault(){}};
    wrapper.instance().handleSubmit(mockEvent);
    expect(filterCards.mock.calls.length).toBe(1);
  });

});
