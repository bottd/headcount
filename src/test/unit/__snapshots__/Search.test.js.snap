// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Renders with one text input and 1 button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Search by name"
          value=""
/>,
        <button>
          Search
</button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search by name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Search by name"
            value=""
/>,
          <button>
            Search
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search by name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
